vsm

```diff
@@ -147,31 +149,55 @@ def process(state, signal, value):
     state.got_signal(signal, value)
 
 
-def run(state):
+def receive():
```

Comment:
Cool. You are still not replacing the send function for the vsm program, just receive. You need to make the emit function call send and in the default implementation send should print to stdout.